### PR TITLE
[SYCL] Remove ASFixer and DISABLE_INFER_AS support

### PR DESCRIPTION
**Pull request by romanovvlad**
Signed-off-by: Vlad Romanov <vlad.romanov@intel.com>

### CODE REVIEW

**Comment by bader** [comment]
@keryell, FYI.

**Comment by agozillon** [comment]
> 
> 
> @keryell, FYI.

It's not needed anymore in the triSYCL project (we still call the pass, but it's redundant and not required), I swapped it to the recommended method a couple of months ago and left a message here: https://github.com/intel/llvm/issues/361 I should have probably been more clear and stated we no longer need the ASFixer/Env variable at the time though... sorry about that. https://github.com/intel/llvm/issues/361 can probably be closed as well.

File: clang/test/CodeGenSYCL/bool-vectors.cpp [hunk 1]
```diff
@@ -1,75 +0,0 @@
-// RUN: %clang_cc1 -triple spir64-unknown-linux-sycldevice  -std=c++11 -fsycl-is-device -disable-llvm-passes -emit-llvm -x c++ %s -o - | opt -asfix -S -o - | FileCheck %s
```
[the remark this answers is not in the document]
Comment (romanovvlad) [reply clang/test/CodeGenSYCL/bool-vectors.cpp]: Because it runs:  `opt -asfix -S -o`

[the remark this answers is not in the document]
Comment (asavonic) [reply clang/test/CodeGenSYCL/bool-vectors.cpp]: Then can you remove `opt -asfix` part and keep the run line?

[the remark this answers is not in the document]
Comment (Fznamznon) [reply clang/test/CodeGenSYCL/bool-vectors.cpp]: I have no idea why it runs asfixer, but the test doesn't seem address space specific, so I suggest do as @asavonic says, remove `opt -asfix` and keep other stuff.

File: clang/test/CodeGenSYCL/address-space-parameter-conversions.cpp [hunk 2]
```diff
@@ -1,201 +0,0 @@
-// RUN: DISABLE_INFER_AS=1 %clang_cc1 -triple spir64-unknown-linux-sycldevice  -std=c++11 -fsycl-is-device -disable-llvm-passes -emit-llvm -x c++ %s -o - | opt -asfix -S -o - | FileCheck %s --check-prefixes CHECK,CHECK-OLD
-// RUN: %clang_cc1 -triple spir64-unknown-linux-sycldevice  -std=c++11 -fsycl-is-device -disable-llvm-passes -emit-llvm -x c++ %s -o - | opt -asfix -S -o - | FileCheck %s --check-prefixes CHECK,CHECK-NEW
```
Comment (asavonic) [inline clang/test/CodeGenSYCL/address-space-parameter-conversions.cpp]: Keep the CHECK-NEW lines here.

File: clang/test/CodeGenSYCL/bool-vectors.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (romanovvlad) [reply clang/test/CodeGenSYCL/bool-vectors.cpp]: Yes

[the remark this answers is not in the document]
Comment (romanovvlad) [reply clang/test/CodeGenSYCL/bool-vectors.cpp]: Done

File: clang/test/CodeGenSYCL/address-space-parameter-conversions.cpp (hunk 2 above)
Comment (romanovvlad) [reply clang/test/CodeGenSYCL/address-space-parameter-conversions.cpp]: Done.

**Pull request merged**